incubator/exporter-otlp/src/test/java/io/aklivity/zilla/runtime/exporter/otlp/internal/serializer/OtlpMetricsSeralizerTest.java

```diff
@@ -0,0 +1,255 @@
+
+        List<MetricRecord> metricRecords = List.of(counterRecord, gaugeRecord, histogramRecord);
+
+        List<AttributeConfig> attributes = List.of(
+            new AttributeConfig("service.namespace", "example")
+        );
+
+        OtlpMetricsDescriptor descriptor = mock(OtlpMetricsDescriptor.class);
+        when(descriptor.nameByBinding("counter1", 42)).thenReturn("counter1");
+        when(descriptor.kind("counter1")).thenReturn("sum");
+        when(descriptor.description("counter1")).thenReturn("description for counter1");
+        when(descriptor.unit("counter1")).thenReturn("");
+
+        when(descriptor.nameByBinding("gauge1", 42)).thenReturn("gauge1");
+        when(descriptor.kind("gauge1")).thenReturn("gauge");
+        when(descriptor.description("gauge1")).thenReturn("description for gauge1");
+        when(descriptor.unit("gauge1")).thenReturn("nanoseconds");
+
+        when(descriptor.nameByBinding("histogram1", 42)).thenReturn("histogram1");
+        when(descriptor.kind("histogram1")).thenReturn("histogram");
+        when(descriptor.description("histogram1")).thenReturn("description for histogram1");
+        when(descriptor.unit("histogram1")).thenReturn("bytes");
+
+        OtlpMetricsSerializer serializer = new OtlpMetricsSerializer(metricRecords, attributes,
+            descriptor::kind, descriptor::nameByBinding, descriptor::description, descriptor::unit);
+        serializer.timeStamp(1234567890);
+
+        // WHEN
+        String json = serializer.serializeAll();
+
+        // THEN
+        assertThat(json, equalTo(expectedJson));
+    }
+
+    @Test
+    public void shouldReturnEmpty()
+    {
+        // GIVEN
+        String expectedJson =
+            "{" +
+                "\"resourceMetrics\":[" +
+                    "{" +
+                        "\"resource\":{" +
+                            "\"attributes\":[]" +
+                        "}," +
+                        "\"scopeMetrics\":[" +
+                            "{" +
+                                "\"scope\":{" +
+                                    "\"name\":\"OtlpMetricsSerializer\"," +
+                                    "\"version\":\"1.0.0\"" +
+                                "}," +
+                                "\"metrics\":[]" +
+                            "}" +
+                        "]" +
+                    "}" +
+                "]" +
+            "}";
+        List<AttributeConfig> attributes = List.of();
+        OtlpMetricsDescriptor descriptor = mock(OtlpMetricsDescriptor.class);
+        OtlpMetricsSerializer serializer = new OtlpMetricsSerializer(List.of(), attributes,
+            descriptor::kind, descriptor::nameByBinding, descriptor::description, descriptor::unit);
+        serializer.timeStamp(1234567890);
+
+        // WHEN
+        String json = serializer.serializeAll();
+
+        // THEN
+        assertThat(json, equalTo(expectedJson));
+    }
+}
```

Comment:
This unit test feels too tight to the implementation so it won't remain stable under refactoring.

Prefer an IT that verifies the protocol output so that implementation can be refactored and verified against unmodified IT.

Note: with the IT in place, this unit test can then be removed as the behavior under test is already covered.